### PR TITLE
Halloween 

### PR DESCRIPTION
**Pull request by bunniewormy**
## Status

(pick one)

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

- Temporary texture changes (to revert in next update after halloween is over):
	- world.png (autumn theme)
	- tree, bush, flowers, grain sprites (autumn/halloween theme)
	- cape sprites (now black)
	- water and bubbles (now a bit more greener)
	- water bombs also green
	- lantern is now a skull
	- heart is now candy
	- traders are zombies
	- skygradient.png (halloween mood)
- Modified gregs, they no longer instakill and instead pick you up for a few seconds (code is very, very, very bad, please dont read it)
- Prolonged halloween holiday to last 28 october - 4 november (the greg spawning period / slightly darker day)

### CODE REVIEW

File: Entities/Zombies/Greg/Greg.as [hunk 1]
```diff
@@ -90,10 +94,11 @@ void onInit(CBlob@ this)
 
     Sound::Play("GregCry.ogg", this.getPosition());
     this.server_setTeamNum(255); //greg team
-    //this.addCommandID("legomyego");
+    this.addCommandID("legomyego");
     this.addCommandID("dropstatue");
     this.addCommandID("unstatue");
     this.set_s32("statue time", 0);
+    this.Sync("statue time", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:101]: Sync not needed, everybody will run this code when the blob is made.

File: Entities/Zombies/Greg/Greg.as [hunk 2]
```diff
@@ -132,12 +137,12 @@ void onTick(CBlob@ this)
             s32 statuetime = this.get_s32("statue time");
             if (statuetime == 0)
             {
-                this.set_s32("statue time", getGameTime() + 45);
+                this.set_s32("statue time", getGameTime() + 1);
+                this.Sync("statue time", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:141]: Sync not needed

File: Entities/Zombies/Greg/Greg.as [hunk 3]
```diff
@@ -149,14 +154,15 @@ void onTick(CBlob@ this)
...
-                map.server_DestroyTile(pos, dmg, this);
+                map.server_DestroyTile(pos, dmg, this);*/
 
             }
             else
             {
                 if (statuetime < getGameTime())
                 {
                     this.set_s32("statue time", 0);
+                    this.Sync("statue time", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:165]: Sync not needed

File: Entities/Zombies/Greg/Greg.as [hunk 4]
```diff
@@ -429,6 +487,17 @@ void onTick(CBlob@ this)
...
 }
 
+void onDie( CBlob@ this )
+{
+    CBlob@ pickedPlayerBlob = this.getAttachmentPoint(0).getOccupied();
+
+    if (pickedPlayerBlob !is null)
+    {
+        pickedPlayerBlob.Untag("picked");
+        pickedPlayerBlob.Sync("picked", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:497]: Sync not needed

File: Entities/Zombies/Greg/Greg.as [hunk 5]
```diff
@@ -378,12 +434,14 @@ void onTick(CBlob@ this)
             {
                 //detach and add some velocity so they hopefully die.
                 this.server_DetachAll();
+                pickedPlayerBlob.Untag("picked");
+                pickedPlayerBlob.Sync("picked", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:438]: Sync not needed

File: Entities/Zombies/Greg/Greg.as [hunk 6]
```diff
@@ -370,6 +423,9 @@ void onTick(CBlob@ this)
                     this.Chat("i lied.");
                 }
 
+                pickedPlayerBlob.Untag("picked");
+                pickedPlayerBlob.Sync("picked", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:427]: Sync not needed

File: Entities/Zombies/Greg/Greg.as [hunk 7]
```diff
@@ -348,16 +368,49 @@ void onTick(CBlob@ this)
...
+                if ((targetAttached && pickedPlayerBlob !is null) || pickedPlayerBlob !is null) 
+                {
+                    CMap@ map = this.getMap();
+                    f32 distToGround = map.getLandYAtX(gregPos.x/8.0f)*8.0f - gregPos.y;
+                    if (distToGround > 240.0f && !is_emote(this, Emotes::troll) && !this.hasTag("doneparis"))
+                    {
+                        if (this.hasTag("paris"))
+                        {
+                            this.Tag("doneparis");
+                            this.Chat("i lied.");
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:390]: Doesnt work on mp

File: Entities/Zombies/Greg/Greg.as [hunk 8]
```diff
@@ -348,16 +368,49 @@ void onTick(CBlob@ this)
...
+                    {
+                        this.Tag("statue");
+                        this.SendCommand(this.getCommandID("dropstatue"));
+                        this.setVelocity(Vec2f(target.getVelocity().x*0.95, 0));
+                        return;
+                    }
+                }
+                if ((targetAttached && pickedPlayerBlob !is null) || pickedPlayerBlob !is null) 
+                {
+                    CMap@ map = this.getMap();
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:383]: use getMap() instead of this,getMap()

File: Entities/Zombies/Greg/Greg.as [hunk 9]
```diff
@@ -451,36 +520,36 @@ void onCollision( CBlob@ this, CBlob@ blob, bool solid )
...
-    }
+    }*/
 
-    /*CBlob@ target = getTarget(this);
-    if (target is blob)
+    CBlob@ target = getTarget(this);
+    if (target is blob && !target.hasTag("picked"))
     {
+        target.Tag("picked");
+        target.Sync("picked", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:529]: Dont need to sync

File: Entities/Zombies/Greg/Greg.as [hunk 10]
```diff
@@ -451,36 +520,36 @@ void onCollision( CBlob@ this, CBlob@ blob, bool solid )
...
+
+        target.set_s32("pickup time", getGameTime());
+        target.Sync("pickup time", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:532]: And again

File: Entities/Zombies/Greg/Greg.as [hunk 11]
```diff
@@ -451,36 +520,36 @@ void onCollision( CBlob@ this, CBlob@ blob, bool solid )
...
+
+        this.set_s32("statue time", 0);
+        this.Sync("statue time", true);
```
Comment (Vam-Jam) [inline Entities/Zombies/Greg/Greg.as:535]: And again

**Review by Vam-Jam — CHANGES_REQUESTED** [review]
Also look into fixing the screaming when ever you change class, it becomes annoying fairly quickly. Its apart of halloween.as onBlobDie

File: Entities/Zombies/Greg/Greg.as (hunk 7 above)
Comment (bunniewormy) [reply Entities/Zombies/Greg/Greg.as]: removed chat stuff

**Pull request merged**